### ISSUE TITLE
Escape key should exit from seek mode

### ISSUE THREAD

**Issue by mwgg** [cboxdoerfer/ddb_waveform_seekbar#8]
Hi,

Currently, when clicking on the seekbar, the only way to exit from seek mode is to release the mouse button. During playback, this would jump the track to wherever the button has been released. Escape key should exit seek mode if user clicked on the seekbar accidentally during playback.

Thanks!

**Comment by cboxdoerfer** [comment]
Hi,

nice idea. I'll add that in the next couple of days. It's probably also a good idea to disable seeking when the mouse pointer left a certain area around the waveform widget.

**Comment by cboxdoerfer** [comment]
After implementing it I'm not sure if quitting with Esc is such a good idea after all, since this requires the waveform widget to steal keyboard focus (most likely from the playlist widget). So everytime you click the waveform widget your playlist shortcuts won't work anymore until you select it again. I found that very inconvenient and until I find a way to fix that I'll rather not add that feature.

So for now I only added the following: When you entered seek mode and leave the waveform widget seeking aborts after a certain distance - and gets enabled again if you still hold the mouse and enter the widget again. Hope this is also works for you.
